### PR TITLE
Adds Janitor Chem Sprayer upgrade to Janitor Cyborg module

### PR DESCRIPTION
**Pull request by ynot01**
locked behind the advanced robotics and advanced janitor nodes
08uycx6
iVIEQst
MQrZRFm
C82hxz8

:cl:  
rscadd: Added Janitor Chem Sprayer upgrade to Janitor Cyborg module
/:cl:

### CODE REVIEW

**Comment by Bibby0110** [comment]
omg such buff

**Comment by HomocidlePancakes** [comment]
Why tho

**Comment by ynot01** [comment]
to improve the life of a poor cyborg whose sole purpose is to clean shit up after others

**Comment by HomocidlePancakes** [comment]
But he is already better at cleaning then anyone else. Why?

**Comment by ynot01** [comment]
because it makes it easier to clean up at the cost of some med-tier mats and 23,000 research points

**Comment by ynot01** [comment]
because it makes it easier to clean up at the cost of some med-tier mats and 23,000 research points

the only meta borgs right now are engi sec and med

**Comment by HomocidlePancakes** [comment]
>But he is already better at cleaning then anyone else. Why?

>because it makes it easier to clean up

Well, you are just reciting the information that is already the question. I asked you “why are you doing this?”, not “what are you doing?”

>at the cost of some med-tier mats and 23,000 research points

1)I did not ask you for research requirments
2)How are it’s research requirments in any way justify this? You just recited them, without an actual explanation as to why they justify the change.

Also I am not too keen on making things, that were specifically made sparse AF, easily obtainable, especially with such a flimsy reasoning. Kinda kills it tbh. Come up with your own memes instead.

**Comment by Bibby0110** [comment]
I like this, merge when?

It shoots space cleaner, in many ways its like the janitors backpack water sprayer

**Comment by ynot01** [comment]
i legit dont know how to tell you any more reasoning than "it makes it easier to clean"

as for the "sparse item made easily obtainable", it's a cyborg, not a carbon, which justifies powerful items specifically for the module. consider secborg having near limitless disabler and stun baton, or medbot being able to constantly synthesize a variety of incredibly useful chemicals into a hypospray, as well as having an entire surgical toolset in its back pocket, or engineering cyborg having access to an RCD that is dependent on charge instead of actual materials (which can be made infinite using yellow cell)

**Comment by HomocidlePancakes** [comment]
I don't. ERT toys are for ERT, not borgs.

**Comment by HomocidlePancakes** [comment]
>i legit dont know how to tell you any more reasoning than "it makes it easier to clean"

If you can't back up your idea with arguments, then it's invalid. Thats all I can say.

>as for the "sparse item made easily obtainable", it's a cyborg, not a carbon

You see, cyborgs will be present in pretty much every shift. ERT on the other hand, especially the janitorial one appears once in a blue moon. It being a avaiable only to borgs doesn't change anything.

> consider secborg having near limitless disabler and stun baton

Yeah, and it's retarded. Saying "This is clearly bad, but someone else is doing it, thus I can do" is just child level arguments. Remove secborg instead.

>or medbot being able to constantly synthesize a variety of incredibly useful chemicals into a hypospray

Sleepers can do this too, and everyone is fine with it. Makes sense that it should be on par with atleast roundstart medbay equipment. Not to mention that medborg, unlike secborg, can't 100% autonomously perform it's duties.

>as well as having an entire surgical toolset in its back pocket

Why shouldn't a medborg be able to perform surgery? It's kinda the whole purpose of it.

>or engineering cyborg having access to an RCD that is dependent on charge instead of actual materials (which can be made infinite using yellow cell)

List of things that the engineering cyborg can do:
Build walls
Lay pipes
Lay cable
Fix essential machinery (APCs, alarms, ETC)

List of thigs the engiborg can't do:
Build/fix specialized machinery
Set up the engine
Build reinforced walls

There are alot of drawbacks to the "Overpowered" ability of being able to build an infinite amount of walls, so don't try to use it as an argument.

Same can be said about all *usefull* cyborgs by the way. They perform general duties better then humans, but can't perform specialized tasks. The only exception to this rule is the secborg which is just yet another reason as to why it should be removed.

Also you've wrote just one line addressing the issues with your PR(which didn't have any arguments btw) and wrote an entire paragraph of mostly unrelated stuff. Do not try to derail the argument by referencing what at first glance looks related, but after 30 second of thought can be easily discarded as irrelevant.

File: yogstation/code/modules/research/designs/mechfabricator_designs.dm [hunk 1]
```diff
@@ -0,0 +1,8 @@
+/datum/design/borg_upgrade_restart
```
Comment (alexkar598) [inline yogstation/code/modules/research/designs/mechfabricator_designs.dm]: Copypasta failure

File: yogstation/code/modules/research/techweb/all_nodes.dm [hunk 2]
```diff
@@ -119,4 +119,13 @@
...
 	design_ids = list("surgery_brainwashing")
 	research_costs = list(TECHWEB_POINT_TYPE_GENERIC = 7000)
-	export_price = 7000
\ No newline at end of file
+	export_price = 7000
+
+/datum/techweb_node/cyborg_upg_custodial
+	id = "cyborg_upg_custodial"
+	display_name = "Cyborg Upgrades: Custodial"
+	description = "We really should fire those janitors."
```
Comment (alexkar598) [inline yogstation/code/modules/research/techweb/all_nodes.dm]:
```suggestion
	description = "We really should kill and borg those janitors."
```

**Review by alexkar598 — CHANGES_REQUESTED** [review]

File: yogstation/code/game/objects/items/robot/robot_upgrades.dm [hunk 3]
```diff
@@ -0,0 +1,28 @@
...
+		R.module.basic_modules += A
+		R.module.add_module(A, FALSE, TRUE)
+
+/obj/item/borg/upgrade/janispray/deactivate(mob/living/silicon/robot/R, user = usr)
+	. = ..()
+	if(.)
+		for(var/obj/item/reagent_containers/spray/chemsprayer/janitor/A in R.module.modules)
+			R.module.remove_module(A, TRUE)
+
+		var/obj/item/reagent_containers/spray/chemsprayer/janitor/M = new (R.module)
```
Comment (alexkar598) [inline yogstation/code/game/objects/items/robot/robot_upgrades.dm:26]:
```suggestion
		var/obj/item/reagent_containers/spray/cyborg_drying/M = new (R.module)
```

**Review by alexkar598 — CHANGES_REQUESTED** [review]

File: yogstation/code/game/objects/items/robot/robot_upgrades.dm (hunk 3 above)
Comment (alexkar598) [reply yogstation/code/game/objects/items/robot/robot_upgrades.dm:26]: wait,this isn't replacing the drying spray,just get rid of those lines after the remove module

**Comment by TacticalSpacePilot** [comment]
Honestly if this were added I would rather have the spray be on janiborg roundstart versus an upgrade which roboticists rarely make.

**Comment by alexkar598** [comment]
This PR has been denied due to negative response

**Comment by Nickvr628** [comment]
What a shame

**Pull request closed without merging**